presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraTokenSplitManager.java

```diff
@@ -13,131 +13,192 @@
-        {
-            checkArgument(range.rpc_endpoints.size() == range.endpoints.size(), "rpc_endpoints size must match endpoints size");
-            this.range = range;
-            this.keyspace = keyspace;
-            this.columnFamily = columnFamily;
-            this.splitSize = splitSize;
-            this.client = client;
-            this.partitioner = partitioner;
+        checkSizeEstimatesTableExist();
+
+        Statement statement = select("range_start", "range_end", "mean_partition_size", "partitions_count")
+                .from(SYSTEM, SIZE_ESTIMATES)
+                .where(eq("keyspace_name", keyspaceName))
+                .and(eq("table_name", tableName));
+
+        ResultSet result = session.executeWithSession(session -> session.execute(statement));
+        ImmutableList.Builder<SizeEstimate> estimates = ImmutableList.builder();
+        for (Row row : result.all()) {
+            SizeEstimate estimate = new SizeEstimate(
+                    row.getString("range_start"),
+                    row.getString("range_end"),
+                    row.getLong("mean_partition_size"),
+                    row.getLong("partitions_count"));
+            estimates.add(estimate);
         }
 
-        @Override
-        public List<TokenSplit> call()
-                throws Exception
-        {
-            ArrayList<TokenSplit> splits = new ArrayList<>();
-            List<CfSplit> subSplits = client.getSubSplits(keyspace, columnFamily, range, splitSize);
-
-            // turn the sub-ranges into InputSplits
-            List<String> endpoints = range.endpoints;
-            TokenFactory factory = partitioner.getTokenFactory();
-            for (CfSplit subSplit : subSplits) {
-                Token left = factory.fromString(subSplit.getStart_token());
-                Token right = factory.fromString(subSplit.getEnd_token());
-                Range<Token> range = new Range<>(left, right, partitioner);
-                List<Range<Token>> ranges = range.isWrapAround() ? range.unwrap() : ImmutableList.of(range);
-                for (Range<Token> subRange : ranges) {
-                    TokenSplit split = new TokenSplit(factory.toString(subRange.left), factory.toString(subRange.right), endpoints);
-
-                    splits.add(split);
-                }
-            }
-            return splits;
+        return estimates.build();
+    }
+
+    private void checkSizeEstimatesTableExist()
+    {
+        KeyspaceMetadata ks = session.executeWithSession(session -> session.getCluster().getMetadata().getKeyspace(SYSTEM));
+        checkState(ks != null, "system keyspace metadata must not be null");
+        TableMetadata table = ks.getTable(SIZE_ESTIMATES);
+        if (table == null) {
+            throw new PrestoException(NOT_SUPPORTED, "Cassandra versions prior to 2.1.5 are not supported");
```

Comment:
https://github.com/prestodb/presto/pull/4525#issuecomment-182219034 -- turns out the minimum supported version of Presto was already 2.1.6